met-web/src/components/engagement/form/EngagementForm.tsx

```diff
@@ -111,27 +114,35 @@ const EngagementForm = () => {
 
         return Object.values(errors).some((isError: unknown) => isError);
     };
-    const handleCreateEngagement = () => {
```

Comment:
I do see the value that it is more succinct to collapse create engagement and update engagement into one method instead of having repetition in two. However, I find it increases cognitive complexity inside of a method especially if we are using an if statement to determine its behavior.

I am hesitant to go that way following the suggested Functional Rules https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29. Namely "Do one thing"
and "fewer arguments".  That's why I separated them. It becomes easier to understand how each method behaves. There is the cost however to do two buttons with a conditional.

Reply:
Changed